### ISSUE TITLE
What is needed to destructuring a function param?

### ISSUE THREAD

**Issue by diorahman** [babel/eslint-plugin-babel#52]
**eslint-plugin-babel version:** 

3.1.0

**babel-eslint version:** 

5.0.0

**eslint version:**

2.1.0

I have following code.

```js
class Foo {

    async print({a, b, c} = d) {

    }

    async spit() {
        await this.print({});
    }

    droll() {

    }
}

const foo = new Foo();

function zap() {

}

zap(foo);
```

Expected:

No errors

And got: 

```
'd' is not defined.
```

What rules or ecmaFeatures should be enabled to get pass on this?

**Comment by hzoo** [comment]
Not really a issue with the plugin - this should be correct? You are are assigning a destructured default to be `d` which is not defined like it says.

**Comment by diorahman** [comment]
Alright. Thank you!
